cleverhans/model.py

```diff
@@ -87,7 +68,7 @@ def get_params(self):
         Provides access to the model's parameters.
         :return: A list of all Variables defining the model parameters.
         """
-        raise NotImplementedError()
+        return tf.global_variables(self.scope)
```

Comment:
Could you comment on why you changed this line?

Reply:
Same response, let's keep the discussion about this on the first comment to avoid repetition.

Reply:
I think that using the scope to keep track of the parameters is too specific to put in the `Model` base class. We want the `Model` class to be able to wrap any kind of 3rd party code, including code written in other frameworks. Since not every framework will necessarily use scopes to keep track of which model owns each Variable, we shouldn't use that mechanism in the `Model` base class. Wrapping models in some frameworks will now require us to change which scope the variables from those frameworks live in.

Reply:
I thought about it it's tricky decision:
1. if we want to avoid tensorflow in Model, we need to clean other functions (get_prob uses tf.nn.softmax for example, get_layer_names too..
2. say if you want to support Keras or PyTorch, you can overload the methods and nothing is lost (and they already do it anyway).